plaso/parsers/mac_keychain.py

```diff
@@ -500,46 +879,21 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
       UnableToParseFile: when the file cannot be parsed.
     """
     try:
-      db_header = self.KEYCHAIN_DB_HEADER.parse_stream(file_object)
-    except (IOError, construct.FieldError):
+      file_header = self._ReadFileHeader(file_object)
+    except (ValueError, errors.ParseError):
       raise errors.UnableToParseFile('Unable to parse file header.')
 
-    if db_header.signature != self.KEYCHAIN_SIGNATURE:
-      raise errors.UnableToParseFile('Not a MacOS keychain file.')
-
-    if (db_header.major_version != self.KEYCHAIN_MAJOR_VERSION or
-        db_header.minor_version != self.KEYCHAIN_MINOR_VERSION):
-      parser_mediator.ProduceExtractionError(
-          'unsupported format version: {0:s}.{1:s}'.format(
-              db_header.major_version, db_header.minor_version))
-      return
-
-    # TODO: document format and determine if -1 offset correction is needed.
-    table_offsets = self._ReadTableOffsets(parser_mediator, file_object)
-    for table_offset in table_offsets:
-      # Skipping X bytes, unknown data at this point.
-      file_object.seek(table_offset, os.SEEK_SET)
-
-      try:
-        table = self.TABLE_HEADER.parse_stream(file_object)
-      except (IOError, construct.FieldError):
-        parser_mediator.ProduceExtractionError(
-            'unable to parse table structure at offset: 0x{0:08x}'.format(
-                table_offset))
-        continue
-
-      # Table_offset: absolute byte in the file where the table starts.
-      # table.first_record: first record in the table, relative to the
-      #                     first byte of the table.
-      file_object.seek(table_offset + table.first_record, os.SEEK_SET)
+    tables = self._ReadTablesArray(file_object, file_header.tables_array_offset)
 
-      if table.record_type == self.RECORD_TYPE_INTERNET:
-        for _ in range(table.number_of_records):
-          self._ReadEntryInternet(parser_mediator, file_object)
+    table = tables.get(self._RECORD_TYPE_APPLICATION_PASSWORD, None)
```

Comment:
You're calling get() with an int argument here, but the docstring from _ReadTablesArray indicates that the dict has string keys. Please check the docstring for the function, I can't readily tell what it's supposed to be.

Reply:
made some changes.